cli/file_watcher.rs

```diff
@@ -26,17 +27,19 @@ async fn error_handler(watch_future: WatchFuture) {
 }
 
 pub async fn watch_func<F>(
-  watch_paths: &[PathBuf],
+  paths: &[PathBuf],
   closure: F,
 ) -> Result<(), AnyError>
 where
   F: Fn() -> WatchFuture,
 {
+  let (mut watcher, receiver) = new_watcher()?;
+  let receiver = Mutex::new(receiver);
   loop {
     let func = error_handler(closure());
     let mut is_file_changed = false;
     select! {
-      _ = file_watcher(watch_paths) => {
+      _ = watch_paths(paths, &mut watcher, &receiver) => {
```

Comment:
If I do understand correctly I have to only reload receiver? When I did this isn't working, I have to unwatch and watch all path after each change or I am doing something wrong

Reply:
Yes, I meant to create a watcher at the beginning and start watching paths, then only poll/await on the receiver end so that there are no two calls to `watch_paths`

Reply:
Sure. Will do :+1: